src/app/rpmostree-ex-builtin-commit2jigdo.c

```diff
@@ -0,0 +1,1167 @@
+                                cancellable, error))
+    return FALSE;
+
+  /* Maps a content object checksum to a set of "objid", which is either
+   * a basename (if unique) or a full path.
+   */
+  g_autoptr(GHashTable) object_to_objid = g_hash_table_new_full (g_str_hash, g_str_equal,
+                                                                 g_free, (GDestroyNotify)g_hash_table_unref);
+  /* Allocate temporary build state (mostly hash tables) just for this call */
+  { g_autoptr(PkgBuildObjidMap) build = g_new0 (PkgBuildObjidMap, 1);
+    build->package = pkg;
+    build->seen_nonunique_objid = g_hash_table_new_full (g_str_hash, g_str_equal,
+                                                                        g_free, NULL);
+    build->seen_objid_to_path = g_hash_table_new_full (g_str_hash, g_str_equal,
+                                                                      g_free, g_free);
+    build->seen_path_to_object = g_hash_table_new_full (g_str_hash, g_str_equal,
+                                                        g_free, g_free);
+    build->tmpfiles_d_path = g_strconcat ("/usr/lib/tmpfiles.d/pkg-",
+                                          dnf_package_get_name (pkg), ".conf", NULL);
+    if (!build_objid_map_for_tree (self, build, object_to_objid,
+                                   commit_root, cancellable, error))
+      return FALSE;
+  }
+
+  /* Loop over the objects we found in this package */
+  GLNX_HASH_TABLE_FOREACH_IT (object_to_objid, it, char *, checksum,
+                              GHashTable *, objid_set)
+    {
+      /* See if this is a "big" object.  If so, we add a mapping from
+       * size → checksum, so we can heuristically later try to find
+       * "content-identical objects" i.e. they differ only in metadata.
+       */
+      guint32 objsize;
+      if (!query_objsize_assert_32bit (self->repo, checksum, &objsize, error))
+        return FALSE;
+      if (objsize >= BIG_OBJ_SIZE)
+        {
+          /* If two big objects that are actually *different* happen
+           * to have the same size...eh, not too worried about it right
+           * now. It'll just be a missed optimization.  We keep track
+           * of how many at least to guide future work.
+           */
+          if (g_hash_table_replace (self->objsize_to_object, GUINT_TO_POINTER (objsize),
+                                    g_strdup (checksum)))
+            self->duplicate_big_pkgobjects++;
+        }
+
+      if (g_hash_table_contains (self->content_object_to_pkg_objid, checksum))
+        {
+          /* We already found an instance of this, just add it to our
+           * duplicate count as a curiosity.
+           */
+          self->n_duplicate_pkg_content_objs++;
+          g_hash_table_iter_remove (&it);
+        }
+      else if (!g_hash_table_contains (self->commit_content_objects, checksum))
+        {
+          self->n_unused_pkg_content_objs++;
```

Comment:
/* This happens a lot for Fedora Atomic Host today where we disable
           * documentation. But it will also happen if we modify any files in
           * postprocessing.
           */